### PR TITLE
Return EINVAL when alignment is not a power of 2 to match posix standard

### PR DESCRIPTION
**Pull request by JulianATA**
Dealt with todo in posix_memalign.

### CODE REVIEW

**Comment by mzji** [comment]
I don't think `(alignment % 2) != 0` is enough to check whether `alignment` is a power of 2 or not. Considering `alignment` is a `size_t`, which is an unsigned integer type, `((x & (~x + 1)) == x && x != 0)` should be sufficient.

**Comment by JulianATA** [comment]
> I don't think `(alignment % 2) != 0` is enough to check whether `alignment` is a power of 2 or not. Considering `alignment` is a `size_t`, which is an unsigned integer type, `(x != 0 && (x & (~x + 1)) == x)` should be sufficient.

Ah.... I see. I'm too careless to treat a checking of power of 2 as a checking of multiple of 2...

File: src/alloc-override.c [hunk 1]
```diff
@@ -132,10 +132,7 @@ size_t malloc_usable_size(void *p)       MI_FORWARD1(mi_usable_size,p)
 void   cfree(void* p)                    MI_FORWARD0(mi_free, p)
 
 int posix_memalign(void** p, size_t alignment, size_t size) {
-  // TODO: the spec says we should return EINVAL also if alignment is not a power of 2.
-  // The spec also dictates we should not modify `*p` on an error. (issue#27)
-  // <http://man7.org/linux/man-pages/man3/posix_memalign.3.html>
-  if (alignment % sizeof(void*) != 0) return EINVAL;  // no `p==NULL` check as it is declared as non-null
+  if ((alignment % sizeof(void*) != 0 )&&((alignment & (~alignment + 1)) == alignment)) return EINVAL;  // no `p==NULL` check as it is declared as non-null
```
Comment (jserv) [inline src/alloc-override.c]: This line is a bit longer. How about splitting into two or more lines?

Comment (jserv) [reply src/alloc-override.c]: For style consistency, it should be `if ((alignment % sizeof(void*) != 0) && `

Comment (JulianATA) [reply src/alloc-override.c]: Thanks for the suggestion! I split the line, and considered the style consistency.

**Comment by daanx** [comment]
Thank you @JulianATA! I think there is a small mistake still, the test should be `(alignment & (~alignment + 1)) != alignment`  (i.e. opposite). I merged it in with the latest dev branch (now with a `make test` target that tests this (among other things)).

**Pull request closed without merging**